docs/src/dev/provider/gremlin-semantics.asciidoc

```diff
@@ -609,6 +609,29 @@ link:https://github.com/apache/tinkerpop/tree/x.y.z/gremlin-core/src/main/java/o
 link:https://github.com/apache/tinkerpop/tree/x.y.z/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/service/ServiceRegistry.java[ServiceRegistry],
 link:https://tinkerpop.apache.org/docs/x.y.z/reference/#call-step[reference]
 
+[[concat-step]]
+=== concat()
+
+*Description:* Concatenates the incoming String traverser with the input String arguments, and return the joined String.
+If the incoming traverser is a non-String value then an `IllegalArgumentException` will be thrown.
+
+*Syntax:* `concat()` | `concat(String...)` | `concat(Traversal)`
+
+[width="100%",options="header"]
+|=========================================================
+|Start Step |Mid Step |Modulated |Domain |Range
+|N |Y |N |`Map` |`String`
+|=========================================================
+
+*Arguments:*
+
+* `String...` - If one or more String values are provided, they will be concatenated with the
+incoming traverser. If no argument is provided, the String value from the incoming traverser is returned.
+* `Traversal` - The `Traversal` value must resolve to a `String`.
+
+See: link:https://github.com/apache/tinkerpop/tree/x.y.z/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/ConcatStep.java[source],
+link:https://tinkerpop.apache.org/docs/x.y.z/reference/#concat-step[reference]
+
```

Comment:
Would be good to include a brief *Exceptions* section here similar to the Call step docs. Just a one liner  regarding IllegalArgumentException would be good (also any exceptions related to nulls if applicable).